### DISCUSSION TITLE
How can I save a file to play back from a microphone pcm16 stream?

### DISCUSSION THREAD

**Discussion by lampemw** [llfbandit/record#367] [Q&A]
I'm not sure if I'm doing something wrong, I'm an novice when it comes to audio data. I'm not sure what filetype I should be saving the iOS pcm16 stream as, but nothing I've tried seems to be playable. 

My Recording configuration:
```dart
RecordConfig(
  encoder: AudioEncoder.pcm16bits,
  numChannels: 1,
),
```

And here's the code I'm using to create the stream and save the data to a file. I've tried m4a, wav filetypes.
```dart
Future<void> recordStream(AudioRecorder recorder, RecordConfig config) async {
  setState(() {
    currentAudioFile = null;
  });
  final path = await _getPath();
  final file = File(path);

  stream = await recorder.startStream(config);
  setState(() {
    isRecording = true;
  });
  stream?.listen((data) {
    file.writeAsBytesSync(data, mode: FileMode.append);
  }, onDone: () {
    print('End of stream. File written to $path.');
    setState(() {
      currentAudioFile = file;
    });
  });
}

Future<String> _getPath() async {
  final dir = await getApplicationDocumentsDirectory();
  return p.join(
    dir.path,
    'audio_${DateTime.now().millisecondsSinceEpoch}.wav',
  );
}
```

When I access the file saved to the path in the app documents directory, it is not playable. Any help would be greatly appreciated as I would like to use the stream to display an amplitude animation but also need to send the file afterwards.

**Comment by Trung15010802** [comment]
Do you find a way ?
